### PR TITLE
Polish API surface

### PR DESCRIPTION
**Pull request by sachaarbonel**
- adopt same function names and signatures than js
- missing convenient methods
### TODOS left:
- [x] type track engagement/impressions method Responses
- [x] avoid returning a dio Response
- [x] introduce a new model for replacing MultipartFile (dio type) 
- [x] refreshCDN
- [x] personalizedFeed

### CODE REVIEW

File: packages/stream_feed/lib/src/client/image_storage_client.dart [hunk 1]
```diff
@@ -99,9 +96,23 @@ class ImageStorageClient {
...
+
+  ///Generate a thumbnail for a given image url
+  Future<String?> thumbnail(String url, int w, int h,
+          {CropType? crop = CropType.center,
+          ResizeType resize = ResizeType.clip}) =>
+      _process(url, {
+        'w': w,
+        'h': h,
+        'crop': crop,
+        'resize': resize,
```
Comment (xsahil03x) [inline packages/stream_feed/lib/src/client/image_storage_client.dart:116]:
```suggestion
        'crop': crop.name,
        'resize': resize.name,
```

Comment (xsahil03x) [reply packages/stream_feed/lib/src/client/image_storage_client.dart:116]: Might also need to public the ResizeType and CropType extension. Or better create a separate class for Thumbnail, similar to Resize and Crop.

File: packages/stream_feed/lib/src/client/users_client.dart [hunk 2]
```diff
@@ -54,6 +64,12 @@ class UsersClient {
     return _users.delete(token, id);
   }
 
+  ///  Get the user profile, it includes the follow counts by default
+  Future<User> profile({bool withFollowCounts = true}) async {
+    checkNotNull(userId, 'userId is not defined');
+    return get(userId!, withFollowCounts: withFollowCounts);
```
Comment (xsahil03x) [inline packages/stream_feed/lib/src/client/users_client.dart:70]: I prefer passing userId as a param. If we are asking for userId in the constructor then we won't be able to use this class as a getter in StreamClient.

File: packages/stream_feed/lib/src/core/models/thumbnail.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  /// [Thumbnail] constructor
+  Thumbnail(this._width, this._height,
+      {required this.resizeType, required this.cropTypes})
+      : assert(_width > 0, 'Width should be a positive number'),
+        assert(_height > 0, 'Height should be a positive number');
+
+  final int _width;
+  final int _height;
+  ResizeType resizeType = ResizeType.clip;
+  List<CropType> cropTypes = const [CropType.center];
```
Comment (xsahil03x) [inline packages/stream_feed/lib/src/core/models/thumbnail.dart:15]: Maybe make these fields final and default values can be assigned in the constructor.

Comment (sachaarbonel) [reply packages/stream_feed/lib/src/core/models/thumbnail.dart:15]: I tried but it didn't work with nnbd

Comment (sachaarbonel) [reply packages/stream_feed/lib/src/core/models/thumbnail.dart:15]: oh it worked without the required keyword

**Review by xsahil03x — APPROVED** [review]
LGTM

**Pull request merged**